zebrad/src/components/mempool.rs

```diff
@@ -1,8 +1,89 @@
 //! Zebra mempool.
 
-/// Mempool-related errors.
-pub mod error;
+use std::{
+    collections::HashSet,
+    future::Future,
+    pin::Pin,
+    task::{Context, Poll},
+};
+
+use futures::future::FutureExt;
+use tower::Service;
+
+use zebra_chain::{
+    parameters::Network,
+    transaction::{UnminedTx, UnminedTxId},
+};
+
+use crate::BoxError;
 
 mod crawler;
+mod error;
+mod storage;
+
+#[cfg(test)]
+mod tests;
 
 pub use self::crawler::Crawler;
+pub use self::error::MempoolError;
+
+#[derive(Debug)]
+#[allow(dead_code)]
+pub enum Request {
+    TransactionIds,
+    TransactionsById(HashSet<UnminedTxId>),
+}
+
+#[derive(Debug)]
+pub enum Response {
+    Transactions(Vec<UnminedTx>),
+    TransactionIds(Vec<UnminedTxId>),
+}
+
+/// Mempool async management and query service.
+///
+/// The mempool is the set of all verified transactions that this node is aware
+/// of that have yet to be confirmed by the Zcash network. A transaction is
+/// confirmed when it has been included in a block ('mined').
+#[derive(Clone)]
+pub struct Mempool {
+    /// The Mempool storage itself.
+    ///
+    /// ##: Correctness: only components internal to the [`Mempool`] struct are allowed to
+    /// inject transactions into `storage`, as they must be verified beforehand.
```

Comment:
Nit: 'they' could also mean 'components' here, but that's not what's intended
```suggestion
    /// ##: Correctness: only components internal to the [`Mempool`] struct are allowed to
    /// inject transactions into `storage`, as transactions must be verified beforehand.
```